packages/source-iotsitewise/src/time-series-data/client/batch.spec.ts

```diff
@@ -1,15 +1,231 @@
 import {
-  createEntryBatches,
-  calculateNextBatchSize,
+  createRawLatestEntryBatches,
+  createRawHistoricalEntryBatches,
+  createAggregateEntryBatches,
+  calculateNextRawHistoricalBatchSize,
+  calculateNextAggregatedBatchSize,
   shouldFetchNextBatch,
-  MAX_BATCH_RESULTS,
-  MAX_BATCH_ENTRIES,
   NO_LIMIT_BATCH,
 } from './batch';
+import {
+  MAX_AGGREGATED_DATA_POINTS,
+  MAX_RAW_HISTORICAL_DATA_POINTS,
+  MAX_RAW_LATEST_REQUEST_ENTRIES,
+  MAX_RAW_HISTORICAL_REQUEST_ENTRIES,
+  MAX_AGGREGATED_REQUEST_ENTRIES,
+} from './constants';
+
+describe('createRawLatestEntryBatches', () => {
+  it('buckets entries by maxResults for a given batch', () => {
+    const batches = createRawLatestEntryBatches([
+      {
+        id: '1',
+        maxResults: undefined,
+      },
+      {
+        id: '2',
+        maxResults: 2000,
+      },
+      {
+        id: '3',
+        maxResults: 2000,
+      },
+      {
+        id: '4',
+        maxResults: 10,
+      },
+      {
+        id: '5',
+        maxResults: 2000,
+      },
+    ]);
+
+    expect(batches).toEqual(
+      expect.arrayContaining([
+        [
+          [
+            {
+              id: '1',
+              maxResults: undefined,
+            },
+          ],
+          NO_LIMIT_BATCH,
+        ],
+        [
+          [
+            {
+              id: '2',
+              maxResults: 2000,
+            },
+            {
+              id: '3',
+              maxResults: 2000,
+            },
+            {
+              id: '5',
+              maxResults: 2000,
+            },
+          ],
+          6000,
+        ],
+        [
+          [
+            {
+              id: '4',
+              maxResults: 10,
+            },
+          ],
+          10,
+        ],
+      ])
+    );
+  });
+
+  it('chunks batches that exceed max entry size (128)', () => {
```

Comment:
nit: `128` in test case name can be parametrized with `MAX_RAW_LATEST_REQUEST_ENTRIES`